### ISSUE TITLE
Implement pyplot.box3d() for Axes3D

### ISSUE THREAD

**Issue by iuryt** [matplotlib/matplotlib#19310]
### Problem

@cgentemann and I were discussing about implementing a new feature for `matplotlib` for a 3D plot formed by three separate 2D surface contour plots, like a box. She thinks `box3d` is a nice name for this feature to conform with matplotlib naming.

Here is a sample image of what I am asking for
![image](https://user-images.githubusercontent.com/5797727/104852008-49365b00-58c6-11eb-83c5-9a2413a69fd3.png)
![image](https://user-images.githubusercontent.com/5797727/104852539-9a941980-58c9-11eb-821c-d76568ac3a63.png)

### Proposed Solution

The example of the implementation is the code in
https://github.com/iuryt/Panthalassan/blob/main/Tutorials/box3D.ipynb
and
https://github.com/cgentemann/cloud_science/blob/master/Butterfly/ecco_3d_eddy_image.ipynb

The idea is to have a way to do the same plot just doing 

```
fig = plt.figure(figsize=(7,4))
ax = fig.add_subplot(111,projection='3d')
ax.box3d.pcolor(X,Y,Z,data,surfaces=[surfacex,surfacey,surfacez],offsets=[offsetx,offsety,offsetz],**kw)
#or
ax.box3d.contour(X,Y,Z,data,surfaces=[surfacex,surfacey,surfacez],offsets=[offsetx,offsety,offsetz],**kw)
#or
ax.box3d.contourf(X,Y,Z,data,surfaces=[surfacex,surfacey,surfacez],offsets=[offsetx,offsety,offsetz],**kw)
```

The default values for the offsets are the same as surfaces, but the user may want to plot a surface for a different offset.
The `kw` arguments are the same as for the used function.

**Comment by jklymak** [comment]
I think an example like that would be super welcome on the gallery page.  It very nicely shows off the abilities of mplot3d.  

I am not 100% sure I understand the need for the `box3d` abstraction.  Maybe I don't know what you mean by the `surfaces` and `offsets` arguments. Nor do I understand how you are instantiating the `box3d` object in the first place.   Is this all really any simpler than just drawing the three surfaces?    I mean the core of this is just 6 lines, right?  

```python
X,Y=np.meshgrid(xx,yy)
ax.contourf(X,Y,z_cut.data,60,zdir='z',offset=zz[0],cmap=cmap,vmin=vmin,vmax=vmax)
Y,Z=np.meshgrid(yy,zz)
ax.contourf(x_cut.data,Y,Z,60,zdir='x',offset=xx[-1],cmap=cmap,vmin=vmin,vmax=vmax) 
X,Z=np.meshgrid(xx,zz)
ax.contourf(X,y_cut.data,Z,60,zdir='y',offset=yy[0],cmap=cmap,vmin=vmin,vmax=vmax)
```

Maybe I am missing subtleties.

**Comment by iuryt** [comment]
@jklymak You are right, is that simple. Surfaces are actually the way I called the cuts from the data and offset is the same as the one used in your example. I think that it is better to put them separately because one might want to plot, for instance, the z=10 cut at z=0.. or something like that.

How can I add my example to the gallery page?

**Comment by jklymak** [comment]
You can add it to the examples at https://github.com/matplotlib/matplotlib/tree/master/examples/mplot3d

> The problem is that everytime I need to plot all surfaces separately, adjust xlim,ylim,zlim and plot the corners and I thought it would be nice to have a function that already does that..

That makes sense - I'm just wondering if abstracting that into its own API makes sense, or if it is just as easy to have a ready-made example?  How about we try the example first - see if it warrants its own API?  We can always update the example later with the new API, so it is not wasted effort.

**Comment by iuryt** [comment]
I agree.

**Comment by rcomer** [comment]
Looks like this was addressed by #19363.  Thanks @saranti for the pointer.
